ledger/participant-integration-api/src/test/suite/scala/platform/store/dao/DeduplicationKeyMakerSpec.scala

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.platform.store.dao
+
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.wordspec.AnyWordSpec
+import com.daml.ledger.api.domain.CommandId
+import com.daml.lf.data.Ref
+
+import java.util.UUID
+import scala.util.Random
+import scalaz.syntax.tag._
+
+class DeduplicationKeyMakerSpec extends AnyWordSpec with Matchers {
+
+  val commandId: CommandId = CommandId(Ref.LedgerString.assertFromString(UUID.randomUUID.toString))
+
+  "DeduplicationKeyMaker" should {
+    "make a deduplication key starting with a command ID in plain-text" in {
+      DeduplicationKeyMaker.make(commandId, List(aParty())) should startWith(commandId.unwrap)
+    }
+
+    "make a deduplication key with a limited length for a large number of submitters" in {
+      val submitters = (1 to 50).map(_ => aParty()).toList
+      val MaxKeyLength = 200
```

Comment:
This constant seems arbitrary, but the value is fine. It's smaller than the maximum length for indexed string columns in both Postgres and Oracle (which was the reason for this change). Maybe add a comment?

Reply:
Good point - I'll comment on motivation for this limit.